shared/src/balancer/event_handler.rs

```diff
@@ -30,24 +32,24 @@ pub struct PoolCreated {
     pub pool_address: H160,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Hash)]
-pub struct WeightedPool {
-    pool_id: H256,
-    pool_address: H160,
-    tokens: Vec<H160>,
-    normalized_weights: Vec<U256>,
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct RegisteredWeightedPool {
+    pub(crate) pool_id: H256,
+    pub(crate) pool_address: H160,
+    pub(crate) tokens: Vec<H160>,
+    pub(crate) normalized_weights: Vec<U256>,
     block_created: u64,
```

Comment:
Why not have everything pub? This seems like the standard "just data" struct where everything could be pub because we return it from other public functions in this file.